### DISCUSSION TITLE
Is Microcks importer able to propagate labels onto Services ?

### DISCUSSION THREAD

**Discussion by GuillaumeGONANO** [microcks/microcks#677] [Q&A]
Hello

We have a use case where we'd like the Microcks importer to :
 1. import in the same "move" (importer) the Mock definitions from the Git repository into Microcks
 2. AND apply the importer's label(s) onto the imported entry in the Microcks "API & Services" list

Currently we can use this workaround :
 1. Have the "basic" importer for the Mock definitions
 2. PLUS have a secondary importer for metadata, containing label(s) in the .yaml file, to apply onto the imported Mock
(as per documentation here : https://microcks.io/documentation/using/advanced/metadata/)

For obvious reasons (ease of use, high number of importer, maintainability, ...) we would prefer this done in a single shot/importer.

Thanks

**Comment by lbroudoux** [comment]
Hey! 

As of today, the labels you put in importers are not propagated directly to the discovered Services during the import. This because it may be a 1-Many relationship and also because no one asked for it yet 😉 

That said - depending on the kind of artifact you're having - you may have the application of labels on your Service if you're using [our OpenAPI](https://microcks.io/documentation/using/openapi/#using-openapi-extensions) or [AsyncAPI](https://microcks.io/documentation/using/asyncapi/#using-asyncapi-extensions) extensions.

If using Postman Collection, unfortunately, there's no way today to specify additional labels to apply. Maybe Postman Collection format supports such things? If knowledgeable on this, please share it with us...
